xrayvision/vis2map.py

```diff
@@ -0,0 +1,239 @@
+
+    Parameters
+    ----------
+    image_dim : tuple
+        (Number of pixels in x, number of pixels in y), default is (64, 64)
+
+    Returns
+    -------
+    mapindex : np.array
+        Two dimensional numpy array which is containing x, y coordinates
+        relative to the image center
+
+    See Also
+    --------
+
+    Notes
+    -----
+
+    Reference
+    ----------
+    | https://darts.isas.jaxa.jp/pub/ssw/gen/idl/image/pixel_coord.pro
+    """
+    data = np.ones(image_dim)
+    mapindex = np.asarray(np.nonzero(data))
+    npixel = np.count_nonzero(data)
+    mapindex[0] = np.subtract(mapindex[0], float(image_dim[0]-1.) / 2.)
+    mapindex[1] = np.subtract(mapindex[1], float(image_dim[1]-1.) / 2.)
+    return mapindex
+
+
+def vis_bpmap_get_xypi(npx: int, pixel):
+    """
+    Calculate sthe spatial weights for the visibilities
+
+    Parameters
+    ----------
+    npx : int
+        x size of the image
+    pixel : float
+
+    Returns
+    -------
+    Spatial Frequency weight
+
+    See Also
+    --------
+
+    Notes
+    -----
+
+    Reference
+    ----------
+    | https://darts.isas.jaxa.jp/pub/ssw/gen/idl/image/vis/vis_bpmap.pro
+    """
+    global xypi_com
+    global xypi
+    global npx_sav
+    global pixel_sav
+    if 'xypi_com' not in globals():
+        xypi_com = 0.
+        xypi = 0.
+        npx_sav = 0.
+        pixel_sav = 0.
+    if (npx != npx_sav) or (pixel_sav != pixel):
+        xypi = np.reshape(pixel_coord((npx, npx)), (2, npx, npx))
+        xypi = np.multiply(xypi, 2. * np.pi * pixel)
+        pixel_sav = float(pixel)
+        npx_sav = int(npx)
+    return xypi
+
+
+def vis_bpmap(visin, bp_fov=80., pixel=0., uniform_weighting=False,
+              spatial_frequency_weight=0):
+    """
+    This procedure makes a backprojection map from a visibility bag
+
+    Parameters
+    ----------
+    visin : np.array
+        np.array of hsi_vis visibility structure (Visibility bag)
+    bp_fov : float
+        field of view (arcsec) Default = 80. arcsec
+    pixel : float
+        size of a pixel
+    uniform_weighting : bool
+        If true, changes subcollimator weighting from default (NATURAL) to
+        UNIFORM. Default if false.
+    spatial_frequency_weight : weighting for each collimator, set by
+        UNIFORM_WEIGHTING if used. The number of weights should either equal
+        the number of unique sub-collimators or the number of visibilities
+
+    Returns
+    -------
+        Backprojection map of the given visibility bag
+
+    See Also
+    --------
+
+    Notes
+    -----
+
+    Reference
+    ----------
+    | https://darts.isas.jaxa.jp/pub/ssw/gen/idl/image/vis/
+    |                   vis_spatial_frequency_weighting.pro
+    """
+    if type(spatial_frequency_weight) == int:
+        spatial_frequency_weight = np.ones((visin.shape[0],))
+    fov = float(bp_fov)
+    pixel = fov / 200.
+    npx = int(fov / pixel)
+    MAP = np.zeros((npx, npx))
```

Comment:
please, do not use upper case variables unless they are not globals in the file. Also, `map` is not a good name as it clashes already with python's `map`  and sunpy's `sunpy.map`.